luigi/worker.py

```diff
@@ -108,10 +108,10 @@ def __init__(self, task, worker_id, result_queue, tracking_url_callback,
         self.result_queue = result_queue
         self.tracking_url_callback = tracking_url_callback
         self.status_message_callback = status_message_callback
-        self.random_seed = random_seed
         if task.worker_timeout is not None:
             worker_timeout = task.worker_timeout
         self.timeout_time = time.time() + worker_timeout if worker_timeout else None
+        self.random_seed = random_seed or self.timeout_time is not None
```

Comment:
Sorry. By set I don't mean `True`/`False`/`Whatever`. I mean that we shouldn't have any default values as they are not used anywhere. That's my claim.

Furthermore. I still don't see anywhere where it's used. Can you convince me that there's a codepath where it's used? From what I see it's not used anywhere.

---

Second read-through. [Here](https://github.com/Houzz/luigi/blob/afe01d32eb8a7fac1249554a8a658393a66a530e/luigi/worker.py#L149-L151) it's referenced. I guess we should rename this variable to `use_random_seed`. As it's not the random seed itself we're providing.